vignettes/epidist.Rmd

```diff
@@ -1,201 +1,250 @@
 
-```{r observe-data}
-truncated_obs <- obs |>
-  filter_obs_by_obs_time(obs_time = 25)
-truncated_obs <- truncated_obs[sample(1:.N, 200, replace = FALSE)]
+```{r}
+all(obs$ptime + obs$delay == obs$stime)
 ```
 
-Plot primary cases (columns), and secondary cases (dots) by the observation time of their secondary events. This reflects would could be observed in real-time.
+If we were to receive the data `obs` as above then estimating the delay distribution would be easy, and the `epidist` package wouldn't need to exist.
+However, in reality, during an outbreak we almost never receive the data as above.
 
-```{r observed-cases}
-truncated_cases <- construct_cases_by_obs_window(
-  obs, windows = c(25), obs_type = "stime"
-)
+First, the times of primary and secondary events will usually be censored.
+This means that rather than exact event times, we observe event times within an interval.
+Here we suppose that the interval is daily, meaning that only the date of the primary or secondary event, not the exact event time, is reported (Figure \@ref(fig:cens)):
 
-plot_cases_by_obs_window(truncated_cases)
+```{r}
+# observe_process() should be renamed and include a "daily" i.e. 1 argument
+obs_cens <- obs |> observe_process()
 ```
 
-We can alternatively plot the observed data based on primary events. This corresponds to a retrospective cohort based view of the data.
-
-```{r cohort-observed-cases}
-truncated_cases <- construct_cases_by_obs_window(
-  obs, windows = c(25), obs_type = "ptime"
-)
+(ref:cens) Interval censoring of the primary and secondary event times obscures the delay times.
 
-plot_cases_by_obs_window(truncated_cases)
+```{r cens, fig.cap="(ref:cens)"}
+ggplot(obs_cens, aes(x = delay, y = delay_daily)) +
+  geom_point(col = "#E69F00") +
+  geom_abline(slope = 1, intercept = 0, linetype = "dashed", col = "grey") +
+  theme_minimal() +
+  coord_fixed() +
+  labs(x = "Exact delay time (days)", y = "Censored delay time (days)")
 ```
 
-Plot the true continuous delay distribution and the empirical observed distribution for each observation window.
+Next, during an outbreak we will usually be estimating delays in real time.
+The result is that only those cases with a secondary event occurring before some time will be observed.
+This is called (right) truncation, and biases the observation process towards shorter delays:
+<!-- I think possible to do a better job with showing what right truncation is with a figure here! -->
 
-```{r empirical-dist}
-combined_obs <- combine_obs(truncated_obs, obs)
+```{r}
+obs_time <- 25
+# filter_obs_by_obs_time() should be renamed to refer to stime
+obs_cens_trunc <- filter_obs_by_obs_time(obs_cens, obs_time = obs_time)
+```
 
-plot_empirical_delay(
-  combined_obs, meanlog = secondary_dist$meanlog[[1]],
-  sdlog = secondary_dist$sdlog[[1]]
-)
+Finally, in reality, it's not possible to observe every case.
+We suppose that a sample of individuals of size `sample_size` are observed:
```

Comment:
if we wanted to make a point about ascertainment here seems like it could be a good choice

Reply:
Happy to leave to you and Katie to decide on this. I'd imagine putting ascertainment into a more complex vignette could be better (especially if there is any analysis the package can do to help you with it.).

Reply:
I think we should make a concise point here and link out to something more complex inn the future